File: tests/inputs/hello.lua
-- Sample Lua file to exercise syntax highlighting, indents,
-- and folds in vorto. Open with `vorto assets/samples/hello.lua`.

local M = {}

local GREETING = "Hello"

--- Build a Person table.
local function new_person(name, age, tags)
    return {
        name = name,
        age = age,
        tags = tags or {},
        is_adult = age >= 18,
    }
end

local function greet(person, prefix)
    prefix = prefix or GREETING
    return string.format("%s, %s!", prefix, person.name)
end

local function classify(n)
    if n < 0 then
        return "negative"
    elseif n == 0 then
        return "zero"
    elseif n % 2 == 0 then
        return "positive even"
    else
        return "positive odd"
    end
end

local function squares(numbers)
    local out = {}
    for _, x in ipairs(numbers) do
        if x * x > 5 then
            out[#out + 1] = x * x
        end
    end
    return out
end

function M.main()
    local alice = new_person("Alice", 30, { "admin", "early_bird" })
    print(greet(alice))

    for n = -2, 5 do
        print(n .. " -> " .. classify(n))
    end

    print("squares: " .. table.concat(squares({ 1, 2, 3, 4, 5 }), ", "))
end

M.main()
return M

File: tests/inputs/hello.rb
# Sample Ruby file to exercise syntax highlighting, indents,
# and folds in vorto. Open with `vorto assets/samples/hello.rb`.

# frozen_string_literal: true

module Demo
  CLASSIFICATIONS = {
    negative: "negative",
    zero: "zero",
    positive_even: "positive even",
    positive_odd: "positive odd"
  }.freeze

  class Person
    attr_reader :name, :age, :tags

    def initialize(name, age, tags = [])
      @name = name
      @age = age
      @tags = tags
    end

    def adult?
      age >= 18
    end
  end

  module_function

  def greet(person, prefix: "Hello")
    "#{prefix}, #{person.name}!"
  end

  def classify(n)
    if n.negative?
      CLASSIFICATIONS[:negative]
    elsif n.zero?
      CLASSIFICATIONS[:zero]
    elsif n.even?
      CLASSIFICATIONS[:positive_even]
    else
      CLASSIFICATIONS[:positive_odd]
    end
  end

  def squares(numbers)
    numbers.map { |x| x * x }.select { |x| x > 5 }
  end
end

alice = Demo::Person.new("Alice", 30, %w[admin early_bird])
puts Demo.greet(alice)

(-2..5).each do |n|
  puts "#{n} -> #{Demo.classify(n)}"
end

puts "squares: #{Demo.squares([1, 2, 3, 4, 5]).join(', ')}"
